imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -40,26 +42,130 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
+                datasets_by_apid[HitAPID.HIT_HSKP], attr_mgr, "imap_hit_l1a_hk"
+            )
         )
-
     if HitAPID.HIT_SCIENCE in datasets_by_apid:
-        # TODO complete science data processing
-        print("Skipping science data for now")
-        datasets_by_apid[HitAPID.HIT_SCIENCE] = process_science(
-            datasets_by_apid[HitAPID.HIT_SCIENCE], attr_mgr
+        l1a_datasets.extend(
+            process_science(datasets_by_apid[HitAPID.HIT_SCIENCE], attr_mgr)
         )
+    return l1a_datasets
 
-    return list(datasets_by_apid.values())
 
+def subcom_sectorates(sci_dataset: xr.Dataset) -> None:
+    """
+    Subcommutate sectorates data.
+
+    Sector rates data contains rates for 5 species and 10
+    energy ranges. This function subcommutates the sector
+    rates data by organizing the rates by species. Which
+    species and energy range the data belongs to is determined
+    by taking the mod 10 value of the corresponding header
+    minute count value in the dataset. A mapping of mod 10
+    values to species and energy ranges is provided in constants.py.
+
+    MOD_10_MAPPING = {
+        0: {"species": "H", "energy_min": 1.8, "energy_max": 3.6},
+        1: {"species": "H", "energy_min": 4, "energy_max": 6},
+        2: {"species": "H", "energy_min": 6, "energy_max": 10},
+        3: {"species": "4He", "energy_min": 4, "energy_max": 6},
+        ...
+        9: {"species": "Fe", "energy_min": 4, "energy_max": 12}}
+
+    The data is added to the dataset as new data fields named
+    according to their species. They have 4 dimensions: epoch
+    energy index, declination, and azimuth. The energy index
+    dimension is used to distinguish between the different energy
+    ranges the data belongs to. The energy min and max values for
+    each species are also added to the dataset as new data fields.
 
-def process_science(dataset: xr.Dataset, attr_mgr: ImapCdfAttributes) -> xr.Dataset:
+    Parameters
+    ----------
+    sci_dataset : xarray.Dataset
+        Xarray dataset containing parsed HIT science data.
     """
-    Will process science dataset for CDF product.
+    # TODO:
+    #  - Update to use fill values defined in attribute manager which
+    #    isn't defined for L1A science data yet
+    #  - fix issues with fe_counts_sectored. The array has shape
+    #      (epoch: 28, fe_energy_index: 1, declination: 8, azimuth: 15),
+    #      but cdflib drops second dimension of size 1 and recognizes
+    #      only 3 total dimensions. Are dimensions of 1 ignored?
+
+    # Calculate mod 10 values
+    hdr_min_count_mod_10 = sci_dataset.hdr_minute_cnt.values % 10
+
+    # Reference mod 10 mapping to initialize data structure for species and
+    # energy ranges and add 8x15 arrays with fill values for each science frame.
+    num_frames = len(hdr_min_count_mod_10)
+    data_by_species_and_energy_range = {
+        key: {**value, "rates": np.full((num_frames, 8, 15), fill_value=-1, dtype=int)}
```

Comment:
Might want to add a TODO to make `dtype` more specific (ex `int16`) once you have a better idea of what that will look like when the CDF attributes are filled out.